src/Symfony/Component/Pgp/PgpEncrypter.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Pgp;
+
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\MailboxListHeader;
+use Symfony\Component\Mime\Message;
+use Symfony\Component\Mime\Part\Multipart\MixedPart;
+use Symfony\Component\Pgp\Exception\BadPassphraseException;
+use Symfony\Component\Pgp\Exception\FileException;
+use Symfony\Component\Pgp\Exception\GeneralException;
+use Symfony\Component\Pgp\Exception\KeyNotFoundException;
+use Symfony\Component\Pgp\Mime\Part\Multipart\PgpEncryptedPart;
+use Symfony\Component\Pgp\Mime\Part\Multipart\PgpSignedPart;
+use Symfony\Component\Pgp\Mime\Part\PgpEncryptedInitializationPart;
+use Symfony\Component\Pgp\Mime\Part\PgpEncryptedMessagePart;
+use Symfony\Component\Pgp\Mime\Part\PgpKeyPart;
+use Symfony\Component\Pgp\Mime\Part\PgpSignaturePart;
+use Symfony\Component\Pgp\Mime\Traits\PgpSigningTrait;
+
+/*
+ * @author PuLLi <the@pulli.dev>
+ */
+final class PgpEncrypter
+{
+    use PgpSigningTrait;
+
+    private \Crypt_GPG $gpg;
+
+    private Headers $headers;
+
+    private ?string $signingKey = null;
+
+    public string $signature = '';
+
+    public string $signed = '';
+
+    /**
+     * @throws FileException
+     * @throws GeneralException
+     */
+    public function __construct(array $options = [])
```

Comment:
We need to describe the available options here.

Alternatively, we might use dependency injection by taking the `Crypt_GPG` object in the constructor

Reply:
Stick with options array, so that projects don't depend directly on the lib, and it can be replaced as needed.

Reply:
Then we should not expose an array of options that is directly the PEAR options either, otherwise we still depend directly on the lib (and we document it in a worse way, making it harder to maintain BC).